frontend/app/services/lookup-service.server.ts

```diff
@@ -531,6 +538,27 @@ function createLookupService() {
     throw new Error(`Failed to fetch data. Status: ${response.status}, Status Text: ${response.statusText}`);
   }
 
+  async function getAllEquityTypes() {
+    const url = `${INTEROP_API_BASE_URI}/lookups/equity-types/`;
+    const response = await fetch(url);
+
+    const equityTypeSchemaList = z.array(equityTypeSchema);
+
+    if (response.ok) {
+      return equityTypeSchemaList.parse(await response.json());
+    }
+
+    log.error('%j', {
+      message: 'Failed to fetch data',
+      status: response.status,
+      statusText: response.statusText,
+      url: url,
+      responseBody: await response.text(),
+    });
+
+    throw new Error(`Failed to fetch data. Status: ${response.status}, Status Text: ${response.statusText}`);
+  }
```

Comment:
**Nitpicky opinion**

The conventional approach to functions like this is to use the following format:

```
const response = fetch(..)
if (!response.ok) {
  // handle error
}
// handle success
```

The reason we do it this way is because it makes the code slightly more maintainable.

Handling error states is usually pretty standard... but post-processing the data coming from an API could contain logic, and change over time. Putting the success state after the error state means that if we update add some post processing in the future, it isn't inside a large `if (..) { .. }` block in the middle of the function.

Reply:
Sure. Implemented in commit https://github.com/DTS-STN/canadian-dental-care-plan/commit/6a3a0642e2595192055662987827b99225e6213b.